packages/vitest/src/node/cli.ts

```diff
@@ -11,6 +11,35 @@ const CLOSE_TIMEOUT = 1_000
 
 const cli = cac('vitest')
 
+type DevCLIConfig = ApiConfig & UserConfig
+interface DevCLIOptions extends DevCLIConfig {}
+
+function buildDevOptions<Options extends DevCLIOptions>(
+  devOptions: Options,
+): Omit<Options, keyof ApiConfig> {
+  const options = { ...devOptions }
+  // @ts-ignore
+  delete options['--']
+  delete options.port
+  delete options.strictPort
+  delete options.host
+  let api: boolean | ApiConfig | undefined
+  if (devOptions.api === true)
+    api = true
+
+  if (devOptions.port !== undefined)
```

Comment:
Maybe we should ignore `port`, `strictPort`, and `host` if `api` is not passed to the cli?

Reply:
The old api option is not necessary, the new one will be always false or objrct, maybe we can add the api option to ResolvedOptions that should be `api?: ApiConfig` and then move again the logic from the node root to the config module

Reply:
Yes, `api` should be in `ResolvedOptions`. My comment was about the cli, if we have `vitest --host`, it should not start the API. I think the user should always pass `vitest --api --host`